rs_server/CADIP/api/cadu_list.py

```diff
@@ -0,0 +1,123 @@
+"""Module for interacting with CADU system through a FastAPI APIRouter.
+
+This module provides functionality to retrieve a list of products from the CADU system for a specified station.
+It includes an API endpoint, utility functions, and initialization for accessing EODataAccessGateway.
+"""
+from datetime import datetime
+from typing import List
+
+from eodag import EOProduct
+from fastapi import APIRouter, status
+from fastapi.responses import JSONResponse
+from services.cadip.rs_server_cadip.cadip_retriever import init_cadip_data_retriever
+from services.common.rs_server_common.data_retrieval.eodag_provider import EodagProvider
+from services.common.rs_server_common.data_retrieval.provider import (
+    CreateProviderFailed,
+)
+
+router = APIRouter()
+
+
+@router.get("/cadip/{station}/cadu/list")
+async def list_cadu_handler(station: str, start_date: str, stop_date: str):
+    """Endpoint to retrieve a list of products from the CADU system for a specified station.
+
+    Parameters
+    ----------
+    station : str
+        Identifier for the CADIP station (MTI, SGS, MPU, INU, etc).
+    start_date : str, optional
+        Start date for time series filter (format: "YYYY-MM-DDThh:mm:sssZ").
+    stop_date : str, optional
+        Stop date for time series filter (format: "YYYY-MM-DDThh:mm:sssZ").
+
+    Returns
+    -------
+    JSONResponse
+        A JSON response containing the list of products (ID, Name) for the specified station.
+        If the station identifier is invalid, a 400 Bad Request response is returned.
+        If no products were found in the mentioned time range, output is an empty list.
+
+    Example
+    -------
+    - Request:
+        GET /cadip/station123/cadu/list?start_date="1999-01-01T12:00:00.000Z"&stop_date="2033-02-20T12:00:00.000Z"
+    - Response:
+        {
+            "station123": [
+                (1, 'Product A'),
+                (2, 'Product B'),
+                ...
+            ]
+        }
+
+    Notes
+    -----
+    - If both start_date and stop_date are provided, products within the specified date range are retrieved.
+    - The response includes a JSON representation of the list of products for the specified station.
+    - In case of an invalid station identifier, a 400 Bad Request response is returned.
+    """
+    if is_valid_format(start_date) and is_valid_format(stop_date):
```

Comment:
You can handle this directly with FastAPI with parameter typing.
Declare your start_date as datetime with isoformat.

Reply:
Agree.